app/components/provider_interface/change_course_details_component.rb

```diff
@@ -9,18 +9,19 @@ class ChangeCourseDetailsComponent < ViewComponent::Base
                 :cycle, :preferred_location, :study_mode, :qualification, :available_providers,
                 :available_courses, :course, :available_course_options, :course_option
 
-    def initialize(application_choice:, course_option: nil, course: nil, available_providers: [], available_courses: [], available_course_options: [])
-      @application_choice = application_choice
+    def initialize(application_choice:, course_option: nil, available_providers: [], available_courses: [], available_course_options: [])
+      @course = application_choice.course_option.course
       @course_option = course_option
-      @provider_name_and_code = application_choice.provider.name_and_code
-      @course_name_and_code = application_choice.course.name_and_code
-      @cycle = application_choice.course.recruitment_cycle_year
+
+      @application_choice = application_choice
+      @provider_name_and_code = course.provider.name_and_code
+      @course_name_and_code = course.name_and_code
+      @cycle = course.recruitment_cycle_year
       @preferred_location = preferred_location_text
-      @study_mode = application_choice.course_option.study_mode.humanize
-      @qualification = qualification_text(application_choice.course_option)
+      @study_mode = course.study_mode.humanize
```

Comment:
we should be reading from the course options study mode not the course itself

Reply:
🆗